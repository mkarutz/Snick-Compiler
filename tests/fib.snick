# Prompts for n, then pints the nth fibonacci number.

proc main()
    int n;
    int m;

    write ">";
    read n;
    fib(n, m);
    write m;
    write "\n";
end

# Procedure fib computes the nth fibonacci number.
# The result is returned in out.

proc fib(val int n, ref int out)
    int f1;
    int f2;

    if n <= 1 then
        out := n;
    else
        fib(n - 2, f1);
        fib(n - 1, f2);
        out := f1 + f2;
    fi
end

