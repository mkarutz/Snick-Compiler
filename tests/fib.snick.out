proc main ()
    int n;
    int m;

    write ">";
    read n;
    fib(n, m);
    write m;
    write "\n";
end

proc fib (val int n, ref int out)
    int f1;
    int f2;

    if n <= 1 then
        out := n;
    else
        fib(n - 2, f1);
        fib(n - 1, f2);
        out := f1 + f2;
    fi
end

